### PR TITLE
Fix Methane Commitment formula & Ecrf file update

### PR DESCRIPTION
**Pull request by isaaccodekill**

### CODE REVIEW

**Review by lemilonkh — COMMENTED** [review]
Thanks ✨

File: app/src/backend/formulas.ts [hunk 1]
```diff
@@ -665,9 +669,12 @@ export async function handleDomesticWasteWaterFormula(
  * @returns The calculated emissions of gases.
  * @throws {createHttpError.BadRequest} If the activity value has no data associated.
  */
+
+// TODO ISSUE WITH WET WASTE NEEDS TO BE FIXED
```
[the remark this answers is not in the document]
Comment (isaaccodekill) [reply app/src/backend/formulas.ts:673]: Yes

File: app/src/backend/formulas.ts [hunk 2]
```diff
@@ -568,11 +572,11 @@ export function handleIndustrialWasteWaterFormula(
...
   const wastewaterGenerated = data[`${prefixKey}-wastewater-generated`];
   const degradableOrganicComponents =
-    data["degradable-organic-components"] ?? 38; // TODO get this from formula values csv;
+    data["degradable-organic-components"] ?? 38; // TODO COD from formula values dependent on industry type;
   const methaneProductionCapacity =
-    data["methane-production-capacity"] ?? DEFAULT_METHANE_PRODUCTION_CAPACITY; // TODO should this only be handled UI-side?
+    data["methane-production-capacity"] ?? DEFAULT_METHANE_PRODUCTION_CAPACITY; // TODO should default to 0.25
   const removedSludge = data["total-organic-sludge-removed"];
-  const methaneCorrectionFactor = 1; // TODO fetch this from formula values csv
+  const methaneCorrectionFactor = 1; // TODO fetch this from formula values csv dependent on treatment type
```
[the remark this answers is not in the document]
Comment (isaaccodekill) [reply app/src/backend/formulas.ts:579]: Yeah a ticket exists

File: app/src/backend/formulas.ts [hunk 3]
```diff
@@ -714,8 +722,7 @@ export async function handleBiologicalTreatmentFormula(
   const totalCH4Emitted = Decimal.mul(organicWasteMass, emissionsFactor).div(
     1000,
   );
-  const totalCH4Recovered =
-    data["biological-treatment-inboundary-total-of-ch4-recovered"] ?? 0;
+  const totalCH4Recovered = data["total-of-ch4-recovered"] ?? 0; // TODO check this.
```
[the remark this answers is not in the document]
Comment (isaaccodekill) [reply app/src/backend/formulas.ts:725]: TODO still needs to remain there until the biological treatment formula is confirmed. Ticket for this exists

File: app/src/components/Tabs/Activity/activity-tab.tsx [hunk 4]
```diff
@@ -149,21 +150,7 @@ const ActivityTab: FC<ActivityTabProps> = ({
...
-      const isActivityDirectMeasure =
-        (activity as unknown as ActivityValue).inventoryValue
-          .inputMethodology === "direct-measure";
-      isDirectMeasure ? isActivityDirectMeasure : !isActivityDirectMeasure;
-    });
-  };
-
-  const activityValues =
-    getActivityValuesByMethodology(getFilteredActivityValues) || [];
+  const activityValues = getFilteredActivityValues;
```
[the remark this answers is not in the document]
Comment (isaaccodekill) [reply app/src/components/Tabs/Activity/activity-tab.tsx:153]: no, it's a memoized value. The name if the variable needs to change though to a better name like "filteredActivityValues"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 5]
```diff
@@ -2043,7 +2043,6 @@
               },
               {
                 "id": "on-road-transport-vehicle-type",
-                "emission-factor-dependency": true,
```
[the remark this answers is not in the document]
Comment (isaaccodekill) [reply app/src/util/form-schema/manual-input-hierarchy.json:2046]: the emission factor dependency key determines which extra fields affect the returned emission factors.  @mfonsecaOEF  clarified that the vehicle type isn't a dependency for sector 2

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 6]
```diff
@@ -3855,6 +3845,10 @@
             ],
             "activity-title": "total-organic-waste-treated",
             "default-units": "units-kilograms",
+            "formula-mapping": {
+              "treatment_type": "biological-treatment-outboundary-treatment-type",
+              "waste_state": "biological-treatment-outboundary-waste-state"
```
[the remark this answers is not in the document]
Comment (isaaccodekill) [reply app/src/util/form-schema/manual-input-hierarchy.json:3850]: Good catch

File: app/src/backend/formulas.ts [hunk 7]
```diff
@@ -568,11 +572,11 @@ export function handleIndustrialWasteWaterFormula(
   const totalIndustrialProduction = data["total-industry-production"];
   const wastewaterGenerated = data[`${prefixKey}-wastewater-generated`];
   const degradableOrganicComponents =
-    data["degradable-organic-components"] ?? 38; // TODO get this from formula values csv;
+    data["degradable-organic-components"] ?? 38; // TODO COD from formula values dependent on industry type;
   const methaneProductionCapacity =
-    data["methane-production-capacity"] ?? DEFAULT_METHANE_PRODUCTION_CAPACITY; // TODO should this only be handled UI-side?
+    data["methane-production-capacity"] ?? DEFAULT_METHANE_PRODUCTION_CAPACITY; // TODO should default to 0.25
```
[the remark this answers is not in the document]
Comment (isaaccodekill) [reply app/src/backend/formulas.ts:577]: yes

**Review by lemilonkh — APPROVED** [review]
Amazing, thanks :raised_hands: :sparkles:

**Pull request merged**